src/User/AutoLoginMiddleware.php

```diff
@@ -1,16 +1,77 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Yiisoft\Yii\Web\User;
 
+use Psr\Http\Server\RequestHandlerInterface;
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Yiisoft\Auth\IdentityInterface;
+use Yiisoft\Auth\IdentityRepositoryInterface;
+use Yiisoft\Yii\Web\User\User;
+
 /**
  * AutoLoginMiddleware automatically logs user in based on "remember me" cookie
  */
-class AutoLoginMiddleware
+final class AutoLoginMiddleware implements MiddlewareInterface
 {
     private User $user;
+    private IdentityRepositoryInterface $identityRepository;
 
-    public function __construct(User $user)
-    {
+    public function __construct(
+        User $user,
+        IdentityRepositoryInterface $identityRepository
+    ) {
         $this->user = $user;
+        $this->identityRepository = $identityRepository;
+    }
+
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
```

Comment:
I think it will be more readable

```php
    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
    {
        if (!$this->userIsAuth($request)) {
            throw new \Exception('Error authentication');
        }

        try {
            $response = $handler->handle($request);
        } catch (\Throwable $e) {
            throw $e;
        }

        return $response;
    }

    private function userIsAuth(ServerRequestInterface $request): bool
    {
        $data = $this->getIdentityAndDurationFromCookie($request);

        if (is_null($data)) {
            return false;
        }

        if (!$this->user->login($data['identity'], $data['duration'])) {
            return false;
        }

        return true;
    }
```

Reply:
Great idea :)

Updated in the last commit